### ISSUE TITLE
Watcher stats missing from diags?

### ISSUE THREAD

**Issue by kostasb** [elastic/support-diagnostics#207]
Looking at the diags.yml https://github.com/elastic/support-diagnostics/blob/master/src/main/resources/diags.yml#L229 the diag tool seems to be running the `GET _watcher/stats/_all` call but in all diag bundles I've gotten so far they don't contain the output of this command. Any idea why it is missing?

Maybe @nemonster knows more?

For reference, these are the output files in the diag bundle (plus the logs directory):

```
alias.json
allocation.txt
allocation_explain.json
allocation_explain_disk.json
cat_aliases.txt
cat_fielddata.txt
cat_health.txt
cat_indices.txt
cat_master.txt
cat_nodes.txt
cat_pending_tasks.txt
cat_recovery.txt
cat_segments.txt
cat_shards.txt
cat_thread_pool.txt
cluster_health.json
cluster_pending_tasks.json
cluster_settings.json
cluster_settings_defaults.json
cluster_state.json
cluster_stats.json
count.json
cpu.txt
cpu_governor.txt
diagnostics.log
dmesg.txt
fielddata.json
fielddata_stats.json
huge_pages.txt
indices_stats.json
iostat.txt
licenses.json
manifest.json
mapping.json
master.json
netstat.txt
nodeattrs.txt
nodes.json
nodes_hot_threads.txt
nodes_stats.json
pipelines.json
plugins.json
process-list.txt
recovery.json
sar.txt
security.properties
segments.json
settings.json
shard_stores.json
shards.json
ss.txt
sysctl.txt
tasks.json
templates.json
top.txt
top_threads.txt
uname.txt
version.json
```

**Comment by jpcarey** [comment]
@kostasb what does the `manifest.json` show for the version of the diagnostic utility?

**Comment by kostasb** [comment]
@jpcarey 

```
{
  "diagToolVersion" : "6.3",
  "inputs" : "InputParams{help=false, outputDir='cwd', host='elk-prd-es01.hogent.be', port=9200, isSsl=false, diagType='diagnostics', reps=1, interval=30, isBzip=false, archivedLogs=false, scrubFiles=false, skipVerification=false, keystore='null', skipLogs=false, skipAccessLogs=false, secured=false, wasPortSet=false}",
  "collectionDate" : "2018-06-27T10:38:03.236+0200"
}
```

**Comment by jpcarey** [comment]
@nemonster `No commercial plugins found - bypassing those REST calls.`
`node: []` <- empty for some reason, so component is not found (test with 6.3.0 elasticsearch)
https://github.com/elastic/support-diagnostics/blob/master/src/main/java/com/elastic/support/diagnostics/commands/RunCommercialQueriesCmd.java#L28

I'll see if I can debug more later, but figured you might have some thoughts there.

**Comment by nemonster** [comment]
There's an issue with the version check that no one noticed/reported. It's
been there a while. It will be fixed on Monday when I release the new one.

Of you need this right away take the diag.yml, copy the calls for the
version you want from the commercial section into the regular call section,
give that to the customer, have them replace the file that came with the
download, and rerun

**Comment by nemonster** [comment]
Just as an example:

If you're running against version 6.x 
then take this section:

```
standard-6:
  alias: "_alias?pretty"
  allocation: "_cat/allocation?v"
  cat_master: "_cat/master?format=json"
  cat_nodes: "_cat/nodes?v&h=ip,heap.percent,ram.percent,cpu,load_1m,load_5m,load_15m,node.role,master,name,nodeId,diskAvail"
  cat_indices: "_cat/indices?v"
  cat_segments: "_cat/segments?v"
  cat_health: "_cat/health?v"
  cat_pending_tasks: "_cat/pending_tasks?v"
  cat_aliases: "_cat/aliases?v"
  cat_thread_pool: "_cat/thread_pool?v"
  cat_fielddata: "_cat/fielddata?v"
  cat_shards: "_cat/shards"
  cluster_health: "_cluster/health?pretty"
  cluster_pending_tasks: "_cluster/pending_tasks?pretty&human"
  cluster_settings: "_cluster/settings?pretty&flat_settings"
  cluster_settings_defaults: "_cluster/settings?include_defaults&pretty&flat_settings"
  cluster_state: "_cluster/state?pretty&human"
  cluster_stats: "_cluster/stats?pretty&human"
  count: "_count?pretty"
  indices_stats: "_stats?pretty&human"
  mapping: "_mapping?pretty"
  master: "_cat/master?format=json"
  nodes_hot_threads: "_nodes/hot_threads?threads=10000"
  nodes_stats: "_nodes/stats?pretty&human"
  fielddata_stats: "/_nodes/stats/indices/fielddata?pretty=true&fields=*"
  fielddata: "_cat/fielddata?format=json&bytes&pretty"
  plugins: "_cat/plugins?format=json"
  cat_recovery: "_cat/recovery?v"
  recovery: "_recovery?pretty"
  shards: "_cat/shards?format=json&bytes=b&pretty"
  segments: "_segments?pretty&human"
  settings: "_settings?pretty&human"
  version: ""
  nodes: "_nodes?pretty&human"
  templates: "_template?pretty"
  nodeattrs: "_cat/nodeattrs"
  tasks: "_tasks?pretty&human&detailed"
  pipelines: "_ingest/pipeline/*?pretty&human"
  allocation_explain: "_cluster/allocation/explain?pretty"
  allocation_explain_disk: "_cluster/allocation/explain?include_disk_info=true&pretty"
  shard_stores: "_shard_stores?pretty"
```
and add these commands from the commercial section:

```
commercial-6:
  licenses: "_license?pretty"
  xpack: "_xpack/usage?pretty&human"
  ml_datafeeds: "_xpack/ml/datafeeds?pretty"
  ml_anomaly_detectors: "_xpack/ml/anomaly_detectors?pretty"
  ml_stats: "_xpack/ml/anomaly_detectors/_stats?pretty"
  watcher_stats: "_watcher/stats/_all"
  security_users: "_xpack/security/user?pretty"
  security_roles: "_xpack/security/role?pretty"
  security_role_mappings: "/_xpack/security/role_mapping?pretty"
```

To make it:

```
standard-6:
  alias: "_alias?pretty"
  allocation: "_cat/allocation?v"
  cat_master: "_cat/master?format=json"
  cat_nodes: "_cat/nodes?v&h=ip,heap.percent,ram.percent,cpu,load_1m,load_5m,load_15m,node.role,master,name,nodeId,diskAvail"
  cat_indices: "_cat/indices?v"
  cat_segments: "_cat/segments?v"
  cat_health: "_cat/health?v"
  cat_pending_tasks: "_cat/pending_tasks?v"
  cat_aliases: "_cat/aliases?v"
  cat_thread_pool: "_cat/thread_pool?v"
  cat_fielddata: "_cat/fielddata?v"
  cat_shards: "_cat/shards"
  cluster_health: "_cluster/health?pretty"
  cluster_pending_tasks: "_cluster/pending_tasks?pretty&human"
  cluster_settings: "_cluster/settings?pretty&flat_settings"
  cluster_settings_defaults: "_cluster/settings?include_defaults&pretty&flat_settings"
  cluster_state: "_cluster/state?pretty&human"
  cluster_stats: "_cluster/stats?pretty&human"
  count: "_count?pretty"
  indices_stats: "_stats?pretty&human"
  mapping: "_mapping?pretty"
  master: "_cat/master?format=json"
  nodes_hot_threads: "_nodes/hot_threads?threads=10000"
  nodes_stats: "_nodes/stats?pretty&human"
  fielddata_stats: "/_nodes/stats/indices/fielddata?pretty=true&fields=*"
  fielddata: "_cat/fielddata?format=json&bytes&pretty"
  plugins: "_cat/plugins?format=json"
  cat_recovery: "_cat/recovery?v"
  recovery: "_recovery?pretty"
  shards: "_cat/shards?format=json&bytes=b&pretty"
  segments: "_segments?pretty&human"
  settings: "_settings?pretty&human"
  version: ""
  nodes: "_nodes?pretty&human"
  templates: "_template?pretty"
  nodeattrs: "_cat/nodeattrs"
  tasks: "_tasks?pretty&human&detailed"
  pipelines: "_ingest/pipeline/*?pretty&human"
  allocation_explain: "_cluster/allocation/explain?pretty"
  allocation_explain_disk: "_cluster/allocation/explain?include_disk_info=true&pretty"
  shard_stores: "_shard_stores?pretty"
  licenses: "_license?pretty"
  xpack: "_xpack/usage?pretty&human"
  ml_datafeeds: "_xpack/ml/datafeeds?pretty"
  ml_anomaly_detectors: "_xpack/ml/anomaly_detectors?pretty"
  ml_stats: "_xpack/ml/anomaly_detectors/_stats?pretty"
  watcher_stats: "_watcher/stats/_all"
  security_users: "_xpack/security/user?pretty"
  security_roles: "_xpack/security/role?pretty"
  security_role_mappings: "/_xpack/security/role_mapping?pretty"
```
Save the new version of the diags.yml, copy it into the unarchived download directory and run again. That should pick up the Watcher stats.

Save process holds for ver 5.

**Comment by kostasb** [comment]
Awesome thanks for figuring out and fixing the root cause @nemonster .

I think that for the time being simply explicitly requesting the watcher stats API call output from the users is easy enough, but I'm glad there's a workaround to have it run as part of the diags.

Looking forward to the new release.

**Comment by kostasb** [comment]
Fixed in v6.4 of the diag tool, closing.
